src/Kirel.Repositories/Infrastructure/Generics/GenericRepository.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System.Linq.Expressions;
+using Kirel.Repositories.Interfaces;
+using Kirel.Repositories.Persistance;
+using Kirel.Repositories.Sorts;
+using Microsoft.EntityFrameworkCore;
+
+namespace Kirel.Repositories.Infrastructure.Generics;
+
+/// <summary>
+/// Generalized Repository class type 
+/// </summary>
+public class GenericRepository<TKey, TEntity> : IGenericRepository<TKey, TEntity> 
+    where TEntity : class, ICreatedAtTrackedEntity, IKeyEntity<TKey> 
+    where TKey :  IComparable, IComparable<TKey>, IEquatable<TKey>
+{
+    private readonly DbContext _dbContext;
+    internal virtual IQueryable<TEntity> _reader
+    {
+        get => _writer.AsQueryable();
+    }
+    internal virtual DbSet<TEntity> _writer
+    {
+        get => _dbContext.Set<TEntity>();
+    }
+    /// <summary>
+    /// GenericRepository constructor
+    /// </summary>
+    /// <param name="context">DbContext</param>
+    public GenericRepository(DbContext context)
+    {
+        _dbContext = context;
+    }
+    /// <summary>
+    /// Add new Entity
+    /// </summary>
+    /// <param name="entity">New entity for creating</param>
+    /// <returns>Created Entity</returns>
+    public async Task<TEntity> Insert(TEntity entity)
+    {
+        _writer.Add(entity);
+        _dbContext.Entry(entity).State = EntityState.Added;
+        await _dbContext.SaveChangesAsync();
+        return entity;
+    }
+    /// <summary>
+    /// Get Entity by ID
+    /// </summary>
+    /// <param name="id">Entity ID</param>
+    /// <returns>Entity</returns>
+    public async Task<TEntity?> GetById(TKey id)
```

Comment:
Return type should be Task<TEntity> (without question mark)

Reply:
Fixed